providers/implementations/keymgmt/hybrid_kmgmt.c

```diff
@@ -0,0 +1,668 @@
+                return 0;                                                   \
+            if (OSSL_PARAM_modified(prm)) {                                 \
+                if (got) {                                                  \
+                    ERR_raise(ERR_LIB_PROV,                                 \
+                              PROV_R_DUPLICATE_VALUES_FOR_GETTER);          \
+                    return 0;                                               \
+                }                                                           \
+                got = 1;                                                    \
+                *param = *prm;                                              \
+            }                                                               \
+        }                                                                   \
+        return 1;                                                           \
+    }
+
+GET_SUBALG_PARAM(key, EVP_PKEY)
+GET_SUBALG_PARAM(ctx, EVP_PKEY_CTX)
+
+#define SET_SUBALG_PARAM(lname, type)                                       \
+    static int lname ## _subalg_set_param                                   \
+        (unsigned int n, type **s, const OSSL_PARAM *param)                 \
+    {                                                                       \
+        OSSL_PARAM prms[2] = { OSSL_PARAM_END, OSSL_PARAM_END };            \
+        unsigned int i;                                                     \
+                                                                            \
+        *prms = *param;                                                     \
+        for (i = 0; i < n; i++) {                                           \
+            if (!type ## _set_params(s[i], prms))                           \
+                return 0;                                                   \
+        }                                                                   \
+        return 1;                                                           \
+    }
+
+SET_SUBALG_PARAM(key, EVP_PKEY)
+SET_SUBALG_PARAM(ctx, EVP_PKEY_CTX)
+
+/* Return the key size and populate the sub-key lengths array */
+#define GET_KEY_LENGTH(lname, type)                                         \
+    uint32_t ossl_hybrid_get_key_length_from_ ## lname(type *s)             \
+    {                                                                       \
+        if (s == NULL)                                                      \
+            return 0;                                                       \
+        if (s->key_lengths[MAX_HYBRID_ALGS] == 0)                           \
+            s->key_lengths[MAX_HYBRID_ALGS] =                               \
+                accumulate_ ## lname ## _uint32_params                      \
+                    (s->num_algs, s->lname ## s,                            \
+                     OSSL_PKEY_PARAM_MAX_SIZE, s->key_lengths);             \
```

Comment:
We're calling this "key lengths" but the param is `OSSL_PKEY_PARAM_MAX_SIZE` which seems odd

Reply:
Yeah, bad nomenclature when the param name was chosen but it's too late to go back and fix it.